cdm/src/main/java/ucar/nc2/iosp/netcdf3/N3header.java

```diff
@@ -682,8 +682,8 @@ void writeHeader(int extra, boolean largeFile, boolean keepDataStart, Formatter
         if (largeFile)
           raf.writeLong(pos);
         else {
-          if (pos > Integer.MAX_VALUE)
-            throw new IllegalArgumentException("Variable starting pos="+pos+" may not exceed "+ Integer.MAX_VALUE);          
+          if (pos > Long.MAX_VALUE)
+            throw new IllegalArgumentException("Variable starting pos="+pos+" may not exceed "+ Long.MAX_VALUE);

```

Comment:
Was this change necessary to support subgroup writing?

Reply:
Our team rewrite some grib2 files(file size is bigger than 110MB). This can throw the exception and I see the pos type is long. So I make this change. The output nc files is bigger than 550MB.

Reply:
Ah, ok. So your use case is converting grib2 to netCDF using the `netcdf4/src/main/java/ucar/nc2/rewrite/Rewrite.java`  class?

Reply:
Yes, our code as follow
![image](https://user-images.githubusercontent.com/5444416/130807483-0248475f-053e-457d-81d9-9d0284a0d0d1.png)

Reply:
Import code:
import ucar.nc2.rewrite.Rewrite;

Reply:
Dependency library:
![Uploading image.png…]()

Reply:
Ok, if that's your goal, then try using [ucar.nc2.FileWriter2](https://docs.unidata.ucar.edu/netcdf-java/4.6/javadoc/ucar/nc2/FileWriter2.html). For larger GRIB files, you will likely not be able to use the `netcdf3` format. We have a pure Java writer for the netCDF 3 classic format, which has a limitation of only supporting 32-bit file offsets. I would suggest writing to one of the netCDF-4 formats (`NetcdfFileWriter.Version.netcdf4_classic` or `NetcdfFileWriter.Version.netcdf4`), which would look something something like this:

~~~java
public static void GribToNc(String filePath) throws IOException {
  File file = new File(filePath);
  NetcdfFile gribFile = NetcdfFile.open(file.getPath());
  String ncFilePath = file.getParent() + File.separator +
          file.getName().substring(0, file.getName().lastIndexOf(".")) +
                  ".nc";
  FileWriter2 writer = new FileWriter2(gribFile, ncFilePath, NetcdfFileWriter.Version.netcdf4, null);
  writer.write();
}
~~~

Writing a netCDF-4 format file requires having access to the netCDF-C library (built with netCDF-4 support). For information on how to have netCDF-java use the netCDF-C library for writing, please see https://docs.unidata.ucar.edu/netcdf-java/4.6/userguide/reference/netcdf4Clibrary.html#loading_preferred. This page is a little out of date, and you no longer need to build HDF5 or netCDF-C in any special way, but the loading instructions are still valid.

If you are using a build system like maven or gradle, you will need the `cdm`, `netcdf4`, and `grib` artifacts. If you are building with `netcdfAll.jar`, those are included.